src/Model/ModelTrait.php

```diff
@@ -60,19 +63,19 @@ protected function loadData(array $data, $update = false, $replaceCollections =
                     // if the setter's first parameter is an array, we need to check if this is a collection of objects
                     // in this case there should be an "addProperty" method on the class
                     $adder = "add$prop";
-                    if ($param->isArray() && method_exists($this, $adder)) {
+                    if ($param->getType() && $param->getType()->getName() === 'array' && method_exists($this, $adder)) {
```

Comment:
Reckon we need the instanceof check here too. `getName()` only exists on `ReflectionNamedType`

Reply:
As it's checking for an array, should I use is_array rather than instanceof?

Reply:
The code is checking the typehint of a method argument via reflection, so `is_array` wouldn't apply here. `$param` should still be a `ReflectionNamedType` at this point

Reply:
Added instanceof check

Reply:
yeah, but not for the right class. Also `array` is not a class so you can't use `instanceof` on it anyway

remember you're not checking for an array, you're checking the object that `getType()` returns is the correct class

Reply:
Should be correct now, the tests pass after I've removed the array changes

Reply:
Not quite. You need to check the `getName()` returns `"array"`

Reply:
Added the array check back in